From: Chris Coulson <chris.coulson@canonical.com>
Date: Thu, 13 Oct 2022 00:02:26 +0100
Subject: apps: pass -propquery arg to the libctx DRBG fetches

Forwarded: no
Applied-Upstream: no
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/openssl/+bug/2056593

The -propquery argument might be used to define a preference for which provider
an algorithm is fetched from. Set the query properties for the library context
DRBG fetches as well so that they are fetched with the same properties.
---
 apps/lib/app_libctx.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/apps/lib/app_libctx.c b/apps/lib/app_libctx.c
index fb26e8e4874e..dd56e472fec5 100644
--- a/apps/lib/app_libctx.c
+++ b/apps/lib/app_libctx.c
@@ -6,6 +6,7 @@
  * in the file LICENSE in the source distribution or at
  * https://www.openssl.org/source/license.html
  */
+#include <openssl/rand.h>
 #include "app_libctx.h"
 #include "apps.h"
 
@@ -15,6 +16,10 @@ static const char *app_propq = NULL;
 int app_set_propq(const char *arg)
 {
     app_propq = arg;
+    if (!RAND_set_DRBG_type(app_libctx, NULL, arg, NULL, NULL))
+        return 0;
+    if (!RAND_set_seed_source_type(app_libctx, NULL, arg))
+        return 0;
     return 1;
 }
 
